tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -841,6 +895,15 @@ impl<F, T> Format<F, T> {
         Ok(())
     }
 
+    /// Set the function to make extra fields.
+    #[cfg(feature = "extra-fields")]
+    pub fn with_make_extra_fields(self, make_trace_id: Box<dyn MakeExtraFields>) -> Self {
```

Comment:
I think this function (and the field on the struct) should be moved to the `Json` struct similarly to the options such as `with_current_span`.

If it were here, it's too easy to misuse because it does nothing for all formatters except for json.